pkg/apis/batch/validation/validation_test.go

```diff
@@ -71,29 +71,85 @@ func getValidGeneratedSelector() *metav1.LabelSelector {
 	}
 }
 
+func getValidPodTemplateSpecForGeneratedRestartPolicyNever(selector *metav1.LabelSelector) api.PodTemplateSpec {
+	podTemplateSpec := getValidPodTemplateSpecForGenerated(selector)
+	podTemplateSpec.Spec.RestartPolicy = api.RestartPolicyNever
+	return podTemplateSpec
+}
+
 func getValidPodTemplateSpecForGenerated(selector *metav1.LabelSelector) api.PodTemplateSpec {
 	return api.PodTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
 			Labels: selector.MatchLabels,
 		},
 		Spec: api.PodSpec{
-			RestartPolicy: api.RestartPolicyOnFailure,
-			DNSPolicy:     api.DNSClusterFirst,
-			Containers:    []api.Container{{Name: "abc", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: api.TerminationMessageReadFile}},
+			RestartPolicy:  api.RestartPolicyOnFailure,
+			DNSPolicy:      api.DNSClusterFirst,
+			Containers:     []api.Container{{Name: "abc", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: api.TerminationMessageReadFile}},
+			InitContainers: []api.Container{{Name: "def", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: api.TerminationMessageReadFile}},
 		},
 	}
 }
 
 func TestValidateJob(t *testing.T) {
+	validJobObjectMeta := metav1.ObjectMeta{
+		Name:      "myjob",
+		Namespace: metav1.NamespaceDefault,
+		UID:       types.UID("1a2b3c"),
+	}
 	validManualSelector := getValidManualSelector()
 	validPodTemplateSpecForManual := getValidPodTemplateSpecForManual(validManualSelector)
 	validGeneratedSelector := getValidGeneratedSelector()
 	validPodTemplateSpecForGenerated := getValidPodTemplateSpecForGenerated(validGeneratedSelector)
+	validPodTemplateSpecForGeneratedRestartPolicyNever := getValidPodTemplateSpecForGeneratedRestartPolicyNever(validGeneratedSelector)
```

Comment:
I wouldn't create a function for one usage. Can you just put the code of that function here?

Reply:
updated